vsintegration/src/FSharp.Editor/CodeFix/ReplaceWithSuggestion.fs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace rec Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Collections.Immutable
+open System.Threading
+open System.Threading.Tasks
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeFixes
+open Microsoft.CodeAnalysis.CodeActions
+
+[<ExportCodeFixProvider(FSharpCommonConstants.FSharpLanguageName, Name = "ReplaceWithSuggestion"); Shared>]
+type internal FSharpReplaceWithSuggestionCodeFixProvider() =
+    inherit CodeFixProvider()
+    let fixableDiagnosticIds = ["FS0039"; "FS1129"; "FS0495"] |> Set.ofList
+    let maybeString = FSComp.SR.undefinedNameSuggestionsIntro()
+        
+    let createCodeFix (title: string, context: CodeFixContext, textChange: TextChange) =
+        CodeAction.Create(
+            title,
+            (fun (cancellationToken: CancellationToken) ->
+                async {
+                    let! sourceText = context.Document.GetTextAsync()
+                    return context.Document.WithText(sourceText.WithChanges(textChange))
+                } |> CommonRoslynHelpers.StartAsyncAsTask(cancellationToken)),
+            title)
+
+    override __.FixableDiagnosticIds = fixableDiagnosticIds.ToImmutableArray()
+
+    override __.RegisterCodeFixesAsync context : Task =
+        async { 
+            context.Diagnostics 
+            |> Seq.filter (fun x -> fixableDiagnosticIds |> Set.contains x.Id)
+            |> Seq.iter (fun diagnostic ->
+                let message = diagnostic.GetMessage()
+                let splitted = message.Split([|maybeString|], StringSplitOptions.None)
```

Comment:
https://en.m.wiktionary.org/wiki/splitted it says I'm nonstandard or archaic. Guess I can live with that. 

Anyways, will fix tomorrow.